src/crimson/osd/recovery_backend.cc

```diff
@@ -161,7 +161,7 @@ seastar::future<BackfillInterval> RecoveryBackend::scan_for_backfill(
   logger().debug("{} starting from {}", __func__, start);
   auto version_map = seastar::make_lw_shared<std::map<hobject_t, eversion_t>>();
   return backend->list_objects(start, max).then(
-    [this, &start, version_map] (auto&& ret) {
+    [this, start, version_map] (auto&& ret) {
```

Comment:
Yeah, the `PGRecovery::request_primary_scan()` path might be problematic. Thanks for catching this!